samza-api/src/main/java/org/apache/samza/system/StreamSpec.java

```diff
@@ -200,4 +200,8 @@ public String get(String propertyName) {
   public String getOrDefault(String propertyName, String defaultValue) {
     return config.getOrDefault(propertyName, defaultValue);
   }
+
+  public SystemStream getSystemStream() {
```

Comment:
This should be avoided, if possible. Since the systemStream is using the physical name:
1. It won't work properly for local systems like CollectionSystemAdmin because there is no physicalName for that one.
2. The physical name is more of a runtime thing, so if the logical layer is depending on it, even indirectly, it's a violation of the separation of concerns.

If the graph needs an identifier, it should use getId()